lnwallet/interface_test.go

```diff
@@ -2376,6 +2372,79 @@ func testChangeOutputSpendConfirmation(r *rpctest.Harness,
 	}
 }
 
+// testSpendUnconfirmed ensures that when can spend unconfirmed outputs.
+func testSpendUnconfirmed(miner *rpctest.Harness,
+	alice, bob *lnwallet.LightningWallet, t *testing.T) {
+
+	// First we will empty out bob's wallet, sending the entire balance
+	// to alice.
+	bobBalance, err := bob.ConfirmedBalance(0)
+	if err != nil {
+		t.Fatalf("unable to retrieve bob's balance: %v", err)
+	}
+	alicePkScript := newPkScript(t, alice, lnwallet.WitnessPubKey)
+	txFeeRate := chainfee.SatPerKWeight(2500)
+	txFee := btcutil.Amount(28760)
+	output := &wire.TxOut{
+		Value:    int64(bobBalance - txFee),
+		PkScript: alicePkScript,
+	}
+	tx := sendCoins(t, miner, bob, alice, output, txFeeRate, true, 1)
+	txHash := tx.TxHash()
+	assertTxInWallet(t, alice, txHash, true)
+	assertTxInWallet(t, bob, txHash, true)
+
+	// Next we will send a transaction to bob but leave it in an
```

Comment:
before sending Bob the coins here, check that Bob fails to send coins because he has no balance. That way we can be sure the later transaction he sends is funded by the unconfirmed output.

Reply:
Good idea. Updated.

Reply:
I don't see the change?

Reply:
Ahh, bad rebase. Should be back now

Reply:
Yep! 👍